### ISSUE TITLE
SEGV in test suite

### ISSUE THREAD

**Issue by eserte** [merkys/Graph-Nauty#8]
The test suite fails on some of my smokers, some of these with a segmentation fault:
```
...
#   Failed test at t/03_pentagon.t line 21.
#          got: '2'
#     expected: '10'

#   Failed test at t/03_pentagon.t line 22.
#          got: '1'
#     expected: '2'
# Looks like you failed 2 tests of 7.
t/03_pentagon.t ................ 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/7 subtests 
t/04_methanol.t ................ 
Failed 4/4 subtests 
...
#   Failed test at t/07_colored_pentagon.t line 16.
#          got: '5'
#     expected: '3'
t/07_colored_pentagon.t ........ 
Failed 2/2 subtests 
t/08_naive_isomorphism_trap.t .. 
Failed 5/5 subtests 
t/09_isomorphism_traps.t ....... 
Failed 4/4 subtests 
...

Test Summary Report
-------------------
t/03_pentagon.t              (Wstat: 512 Tests: 7 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
t/04_methanol.t              (Wstat: 11 Tests: 0 Failed: 0)
  Non-zero wait status: 11
  Parse errors: Bad plan.  You planned 4 tests but ran 0.
t/07_colored_pentagon.t      (Wstat: 11 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero wait status: 11
  Parse errors: Bad plan.  You planned 2 tests but ran 1.
t/08_naive_isomorphism_trap.t (Wstat: 11 Tests: 0 Failed: 0)
  Non-zero wait status: 11
  Parse errors: Bad plan.  You planned 5 tests but ran 0.
t/09_isomorphism_traps.t     (Wstat: 11 Tests: 0 Failed: 0)
  Non-zero wait status: 11
  Parse errors: Bad plan.  You planned 4 tests but ran 0.
```

**Comment by merkys** [comment]
Thanks for reporting. Could you please specify the version of nauty library used in your smokers?

**Comment by eserte** [comment]
Looking at http://matrix.cpantesters.org/?dist=Graph%3A%3ANauty;os=linux;reports=1 it seems that every Linux distribution running on Smokers (Fedora, Ubuntu, Debian) is affected. Usually the system-provided packages are used; for debian10 this is 2.6r10+ds-1

**Comment by merkys** [comment]
Thanks for the clarification. I will try to reproduce.

**Comment by merkys** [comment]
I managed to reproduce on a machine similar to [this one](http://www.cpantesters.org/cpan/report/8afd34f4-971f-11eb-b0cb-5d491f24ea8f). Apparently, failures occur with [Graph](https://metacpan.org/release/Graph) > 0.9716. Must be some sort of API/invariant change, will work on that.

**Comment by merkys** [comment]
Fixed in da41b08a02d1c9b35b4430b1ed0070031a9f513f. There was a change in Graph API (https://github.com/graphviz-perl/Graph/issues/22), I have rewritten the code to work both with older and newer versions of Graph.

Closing the issue for now, please reopen if persists.
